crates/services/tx_status_manager/src/tx_status_stream.rs

```diff
@@ -86,12 +87,47 @@ impl TxUpdateStream {
         self.state = match state {
             State::Empty => match msg {
                 TxStatusMessage::Status(TransactionStatus::Submitted(s)) => {
-                    State::Initial(TransactionStatus::Submitted(s))
+                    State::Submitted(TransactionStatus::Submitted(s))
                 }
+
+                TxStatusMessage::Status(TransactionStatus::PreConfirmationSuccess(s)) => {
+                    State::Preconfirmed(TransactionStatus::PreConfirmationSuccess(s))
+                }
+                TxStatusMessage::Status(TransactionStatus::PreConfirmationFailure(s)) => {
+                    State::Preconfirmed(TransactionStatus::PreConfirmationFailure(s))
+                }
+                TxStatusMessage::Status(
+                    TransactionStatus::PreConfirmationSqueezedOut(s),
+                ) => {
+                    State::Preconfirmed(TransactionStatus::PreConfirmationSqueezedOut(s))
+                }
```

Comment:
```suggestion
```

We can treat it as `State::EarlySuccess(s)` because the transaction definitely is not included in the block.

Reply:
Done